### PR TITLE
Preventing ID and Type from being editable at wrong times

### PR DESCRIPTION
**Pull request by lfilho**

### CODE REVIEW

**Comment by joshwnj** [comment]
Can this be achieved purely within the render function, by changing the condition on the `readOnly` attribute?

```
          <input ref="idField" name="id" placeholder="ID" autoFocus={true} defaultValue={id} readOnly={!!exists} />
```

Would be good to avoid directly touching the DOM if we can.

**Comment by lfilho** [comment]
Josh, please see the updated diff. I couldn't think of a way of making Type readonly without manipulating the dom directly, since the state is already setted...

File: frontend/components/wiki.js [hunk 1]
```diff
@@ -70,12 +77,15 @@ module.exports = React.createClass({
 
   onClickSave: function (event) {
     event.preventDefault();
+    var self = this;
 
     var id = this.state.id;
     var type = this.refs.typeField.getDOMNode().value.trim();
```
Comment (lfilho) [inline frontend/components/wiki.js]: Should'nt this be `var type = this.state.type;` as well? Since we have `onChangeType` updating the state...

Comment (joshwnj) [inline frontend/components/wiki.js]: Yeah let's come back to that in a cleanup, since a few things will likely change anyway.

**Pull request merged**